core/node/node_framework/src/implementations/layers/web3_api/tx_sender.rs

```diff
@@ -92,6 +93,7 @@ impl WiringLayer for TxSenderLayer {
         }));
 
         // Build `TxSender`.
+        let whitelisted_tokens_for_aa = self.tx_sender_config.whitelisted_tokens_for_aa.clone();
         let mut tx_sender = TxSenderBuilder::new(self.tx_sender_config, replica_pool, tx_sink);
```

Comment:
Nit: Since tokens are a part of `tx_sender_config`, maybe it makes sense to initialize tokens `TxSenderBuilder` using the config and allow to override this value?

```rust
impl TxSenderBuilder {
    pub fn new(config, ..) -> Self {
        // initializes tokens from `config`
    }

    pub fn with_whitelisted_tokens_cache(mut self, cache: Arc<RwLock<_>>) -> Self {
        // sets the cache
    }
}
```